src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -189,22 +305,85 @@ http_flush_response_bytes(
         pfCompletionExpected);
     return hr;
 }
-// End of export 
 
+EXTERN_C __MIDL_DECLSPEC_DLLEXPORT
+HRESULT
+http_enable_websockets(
+    _In_ IHttpContext* pHttpContext
+)
+{
+    if (!g_fWebSocketSupported)
+    {
+        return E_FAIL;
+    }
 
-IN_PROCESS_APPLICATION*  IN_PROCESS_APPLICATION::s_Application = NULL;
+    ((IHttpContext3*)pHttpContext)->EnableFullDuplex();
```

Comment:
I think this will lead to crash on Win7 as the method cannot be found

Reply:
Yeah these need to be put in a separate file eventually.

Reply:
Next PR?

Reply:
yes. I will do some test on win7 and solve it with next PR